sdk/core/azure-core/src/main/java/com/azure/android/core/credential/AccessToken.java

```diff
@@ -14,30 +14,30 @@
 
     /**
      * Creates an access token instance.
-     * @param token the token string.
-     * @param expiresAt the expiration time.
+     * @param token The token string.
+     * @param expiresAt The expiration time.
      */
     public AccessToken(String token, OffsetDateTime expiresAt) {
         this.token = token;
         this.expiresAt = expiresAt.minusMinutes(2); // 2 minutes before token expires
```

Comment:
I know we talked about using this "T-2 minutes" value as the expiration before, but I couldn't for the life of me figure out where that idea originated. I ended up taking it out of the iOS implementation. If we keep `expiresAt` here (which we may not depending on the decision about checking for expiration), we should probably remove this magic 2 minute adjustment to the expiration time.

Reply:
I just talked to Jianghao on the Java team about this since they use the 2 minutes subtraction. He explained that the idea is that there's delay between when an access token's expiry time is checked in the `BearerTokenPolicy` and when the request reaches the service, this is more so when the Netty request queue is backed up and a request is sitting there for a long time. However, they are changing this behavior soon so `AccessToken` returns the actual `expiresAt` value and moving it to `SimpleaTokenCache`.

Based on this. I believe we can do as you suggest and remove the subtraction altogether and worry about the delay between a request being made and it being serviced somewhere else.

Reply:
Interesting, so on iOS we don't look at the expiration at all, we just send the token to the service and let it respond with an error if it's expired. Also we don't currently have any mechanism for caching tokens (which I believe is also true for Android? - correct me if I'm wrong. I will have to look into this caching mechanism, because the logic for caching tokens seemed pretty complex when I started thinking about it.

Reply:
I agree that checking for expiration in the cache makes way more sense than to do it in the token itself. For now, I think removing this check from the token and let the service tell us if it's expired is the best way to go, like you suggested.